### PR TITLE
Feature/batched multiclass

### PR DESCRIPTION
**Pull request by HaukurPall**
- Adding batching for multiclass hub interface
- Removing unsude code in multiclass task.

### CODE REVIEW

File: src/greynirseq/nicenlp/models/multiclass.py [hunk 1]
```diff
@@ -158,7 +159,7 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.word_start_dict = get_word_beginnings(self.args, self.task.dictionary)
 
-    def encode(self, sentence):
+    def encode(self, sentence) -> torch.LongTensor:
```
Comment (vesteinn) [inline src/greynirseq/nicenlp/models/multiclass.py:162]: Sentence is missing type

File: src/greynirseq/nicenlp/models/multiclass.py [hunk 2]
```diff
@@ -169,16 +170,45 @@ def decode(self, tokens):
...
         return super().decode(tokens)[1:]
 
-    def predict_labels(self, sentence):
-        tokens = self.encode(sentence)
-        word_mask = torch.tensor([self.word_start_dict[t] for t in tokens.tolist()])
-        word_mask[0] = 0
-        word_mask[-1] = 0
-        word_mask = word_mask.unsqueeze(0)
-        tokens = tokens.unsqueeze(0)
+    def predict_labels(self, sentences: List[str], batch_size=1) -> Iterable[List[str]]:
```
Comment (vesteinn) [inline src/greynirseq/nicenlp/models/multiclass.py:173]: This returns a generator

File: src/greynirseq/nicenlp/models/multiclass.py [hunk 3]
```diff
@@ -169,16 +170,45 @@ def decode(self, tokens):
...
+            sentences: An list of the rule-based tokenized sentences.
+            batch_size: The number of sentences to process in parallel.
+        Returns:
+            An iterable of the labels.
+        """
+        length = len(sentences)
+        for ndx in range(0, length, batch_size):
+            batch = sentences[ndx : min(ndx + batch_size, length)]
+            labels, pred_idx = self._predict_labels(batch)
+            yield from labels
```
Comment (vesteinn) [inline src/greynirseq/nicenlp/models/multiclass.py:186]: Does this call for a PR in GreynirSeqAPI ?

**Review by vesteinn — CHANGES_REQUESTED** [review]

File: src/greynirseq/nicenlp/models/multiclass.py (hunk 2 above)
Comment (peturorri) [reply src/greynirseq/nicenlp/models/multiclass.py:173]: Generators er iterable (er það ekki?). Held hins vegar að týpan ætti að vera `Iterable[str]` nema ég sé að misskilja yield from.

File: src/greynirseq/nicenlp/models/multiclass.py (hunk 3 above)
Comment (HaukurPall) [reply src/greynirseq/nicenlp/models/multiclass.py:186]: Yes, since the default behavior is to return a Generator[Labels] instead of a single Labels.

File: src/greynirseq/nicenlp/models/multiclass.py (hunk 2 above)
Comment (HaukurPall) [reply src/greynirseq/nicenlp/models/multiclass.py:173]: Já, generators eru iterable en þetta á að vera Iterable[str].

Comment (vesteinn) [reply src/greynirseq/nicenlp/models/multiclass.py:173]: Fair enough, but is there any reason not to use the generator type? https://docs.python.org/3/library/typing.html#typing.Generator

Comment (HaukurPall) [reply src/greynirseq/nicenlp/models/multiclass.py:173]: Mér finnst það í þessu tilfelli vera eins og að skila "int" fyrir fall sem skilar "bool". Að skila Generator[List[str], None, None] finnst mér minna lýsandi.

Comment (vthorsteinsson) [reply src/greynirseq/nicenlp/models/multiclass.py:173]: Það sem ykkur vantar er Iterator[], ekki Generator[].

**Review by vesteinn — APPROVED** [review]
